File: Data Structures & Algorithms/linked-list-cycle-detection/submission-2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        tortise = head
        hare = head
        isFirst = True
 
        while tortise and hare:
            if (not isFirst) and (tortise == hare) :
                return True
            
            tortise = tortise.next
            hare = hare.next
            if hare:
                hare = hare.next
            isFirst = False
        
        return False

File: Data Structures & Algorithms/reverse-a-linked-list/submission-2.swift
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     var val: Int
 *     var next: ListNode?
 *     init(_ val: Int) {
 *         self.val = val
 *         self.next = nil
 *     }
 * }
 */

class Solution {
    func reverseList(_ head: ListNode?) -> ListNode? {
        
        var curr: ListNode? = head
        var prev: ListNode? = nil

        while curr != nil {
            let next = curr!.next

            curr!.next = prev

            prev = curr
            curr = next
        }
        return prev

    }
}
